src/etc/cat-and-grep.sh

```diff
@@ -39,7 +39,7 @@ while getopts ':vieh' OPTION; do
             GREPFLAGS="i$GREPFLAGS"
             ;;
         e)
-            GREPPER=egrep
+            GREPFLAGS="E$(echo "$GREPFLAGS" | cut -c2-)"
```

Comment:
Why are we trimming the first two characters here? That feels like a weird thing to me, not sure I understand it.

We should have a comment in the code explaining it as well.

Reply:
It trims the first character 'F' which represents `fgrep`. I should add a comment to explain that.

Reply:
Is there a reason we can't add GREPFLAGS=F in the other branches or so, rather than trying to *remove* it here? I would love to avoid having to think what the behavior here is across loop iterations, for example.

Honestly, the actual correct thing might be to rewrite this script into some non-shell language, but that goes with a larger rewrite of the runmake tests to avoid make in the first place.

Reply:
Oh, I could use `echo "$GREPFLAGS" | tr -d F` which is more readable and less error prone.

Reply:
> Is there a reason we can't add GREPFLAGS=F in the other branches or so, rather than trying to _remove_ it here? I would love to avoid having to think what the behavior here is across loop iterations, for example.
> 
> Honestly, the actual correct thing might be to rewrite this script into some non-shell language, but that goes with a larger rewrite of the runmake tests to avoid make in the first place.

I think it make sense to use F flag by default since the default grepper is fgrep in the original script.

Reply:
Right, but we can do that with a default branch or something, after the loop perhaps? Rather than removing it specifically in this branch.

Reply:
> Right, but we can do that with a default branch or something, after the loop perhaps? Rather than removing it specifically in this branch.

That's a good idea!